packages/core-data/src/resolvers.js

```diff
@@ -165,8 +165,44 @@ export const getEntityRecord =
 					}
 				}
 
-				const record = await apiFetch( { path } );
-				dispatch.receiveEntityRecords( kind, name, record, query );
+				const response = await apiFetch( { path, parse: false } );
+				const record = await response.json();
+
+				const allowHeader = response.headers?.get( 'allow' );
+				const allowedMethods = allowHeader?.allow || allowHeader || '';
+				const permissions = {};
+				const methods = {
+					create: 'POST',
+					read: 'GET',
+					update: 'PUT',
+					delete: 'DELETE',
+				};
+				for ( const [ actionName, methodName ] of Object.entries(
+					methods
+				) ) {
+					permissions[ actionName ] =
+						allowedMethods.includes( methodName );
+				}
+
+				registry.batch( () => {
+					dispatch.receiveEntityRecords( kind, name, record, query );
+
+					for ( const action of [
+						'create',
+						'read',
+						'update',
+						'delete',
+					] ) {
+						const permissionKey = [ action, kind, name, key ]
+							.filter( Boolean )
+							.join( '/' );
+
+						dispatch.receiveUserPermission(
```

Comment:
I think that's a good idea but we should also mark the resolvers as resolved to avoid further REST API requests when the selectors get called no?

Reply:
Correct, and I think I need to update `canUser` as well.

I see that some resolvers for similar cases call both `startResolution` and `finishResolution` actions, while some only call `finishResolution`. Which method is generally preferred?

Reply:
If you'll do it inside the `batch`, then only `finishResolution` makes sense. `start` + `finish` is useful is when you want the observers to see the "tick", but when calling listeners is disables, as it is in `batch`, then `start` does nothing useful.

Reply:
Thanks, @jsnajdr!